roles/openshift_sanitize_inventory/tasks/main.yml

```diff
@@ -19,10 +25,11 @@
     openshift_release: "{{ openshift_release | string | regex_replace('^v', '') }}"
   when: openshift_release is defined
 
-- name: Ensure a valid deployment type has been given.
-  # this variable is required; complain early and clearly if it is invalid.
-  when: openshift_deployment_type not in known_openshift_deployment_types
+- name: Validate openshift_release looks like a version
+  when: |
+    openshift_release is defined
+    and not openshift_release | match('\d+(\.\d+){1,3}$')
```

Comment:
I wish `assert` weren't so needlessly verbose. Both of these solutions are less than ideal to me. The skipped `fail` task can be vague or ambiguous when reading a log. You can read this as 'ensuring a valid deployment type was given' was skipped. As in, no verification happened, because it was skipped.

I wish assert didn't need to print out all that garbage.

With the fail though I think the title could be adjusted to make it read clearer

Original
```
TASK [openshift_sanitize_inventory : Ensure a valid deployment type has been given] ***
skipping: [localhost]
```
^ Reads to me as "We skipped checking this"


Revision
```
TASK [openshift_sanitize_inventory : Abort if an invalid deployment type has been given] ***
skipping: [localhost]
```
^ Reads to me as "We checked this, you're good, we're going to skip aborting"